Libraries/LibWeb/CSS/Parser/ValueParsing.cpp

```diff
@@ -791,6 +795,107 @@ RefPtr<CSSStyleValue const> Parser::parse_percentage_value(TokenStream<Component
     return nullptr;
 }
 
+// https://drafts.csswg.org/css-anchor-position-1/#sizing
+RefPtr<AnchorSizeStyleValue const> Parser::parse_anchor_size(TokenStream<ComponentValue>& tokens)
+{
+    // anchor-size() = anchor-size( [ <anchor-name> || <anchor-size> ]? , <length-percentage>? )
+
+    auto transaction = tokens.begin_transaction();
+    auto const& function_token = tokens.consume_a_token();
+    if (!function_token.is_function("anchor-size"sv))
+        return {};
+
+    // It is only allowed in the accepted @position-try properties (and is otherwise invalid).
+    static Array allowed_property_ids = {
+        // inset properties
+        PropertyID::Inset,
+        PropertyID::Top, PropertyID::Right, PropertyID::Bottom, PropertyID::Left,
+        PropertyID::InsetBlock, PropertyID::InsetBlockStart, PropertyID::InsetBlockEnd,
+        PropertyID::InsetInline, PropertyID::InsetInlineStart, PropertyID::InsetInlineEnd,
+        // margin properties
+        PropertyID::Margin,
+        PropertyID::MarginTop, PropertyID::MarginRight, PropertyID::MarginBottom, PropertyID::MarginLeft,
+        PropertyID::MarginBlock, PropertyID::MarginBlockStart, PropertyID::MarginBlockEnd,
+        PropertyID::MarginInline, PropertyID::MarginInlineStart, PropertyID::MarginInlineEnd,
+        // sizing properties
+        PropertyID::Width, PropertyID::MinWidth, PropertyID::MaxWidth,
+        PropertyID::Height, PropertyID::MinHeight, PropertyID::MaxHeight,
+        PropertyID::BlockSize, PropertyID::MinBlockSize, PropertyID::MaxBlockSize,
+        PropertyID::InlineSize, PropertyID::MinInlineSize, PropertyID::MaxInlineSize,
+        // self-alignment properties
+        PropertyID::AlignSelf, PropertyID::JustifySelf, PropertyID::PlaceSelf,
+        // FIXME: position-anchor
+        // FIXME: position-area
+    };
+    bool valid_property_context = false;
+    for (auto& value_context : m_value_context) {
+        if (!value_context.has<PropertyID>())
+            continue;
+        if (!allowed_property_ids.contains_slow(value_context.get<PropertyID>())) {
+            valid_property_context = false;
+            break;
+        }
+        valid_property_context = true;
+    }
+    if (!valid_property_context)
+        return {};
+
+    auto context_guard = push_temporary_value_parsing_context(FunctionContext { function_token.function().name });
+    auto argument_tokens = TokenStream { function_token.function().value };
+
+    Optional<FlyString> anchor_name;
+    Optional<Keyword> anchor_size;
+    Variant<Empty, LengthPercentage, NonnullRefPtr<AnchorSizeStyleValue const>> fallback_value;
```

Comment:
I'm not 100% sure, but I think it may be better to parse the `LengthPercentage` as a `LengthPercentageStyleValue`, then `fallback_value` could be a `RefPtr<CSSStyleValue const>`. I think you might see an advantage when serializing calculated values in `AnchorSizeStyleValue`.